### PR TITLE
Sscs 9747 ready to list or not listable

### PR DESCRIPTION
**Pull request by gmikerlemos**
Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-9747

### Change description ###

Child Support: Set Case to Ready to List or Not Listable

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java [hunk 1]
```diff
@@ -43,6 +45,28 @@ public static void assignOtherPartyId(List<CcdValue<OtherParty>> otherParties) {
...
+    }
+
+    private static boolean hasDueDateSetAndOtherPartyWithoutHearingOption(SscsCaseData sscsCaseData) {
+        return StringUtils.isNotBlank(sscsCaseData.getDwpDueDate())
+                && !everyOtherPartyHasAtLeastOneHearingOption(sscsCaseData);
+    }
+
+    private static boolean everyOtherPartyHasAtLeastOneHearingOption(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOtherParties() != null) {
+            return sscsCaseData.getOtherParties().stream().noneMatch(otherParty -> otherParty.getValue().getHearingOptions() == null);
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:64]: Is this correct?

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java [hunk 2]
```diff
@@ -43,6 +45,28 @@ public static void assignOtherPartyId(List<CcdValue<OtherParty>> otherParties) {
...
+        } else {
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:65]: less lines if you remove the else and leave the return false

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java (hunk 1 above)
Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:64]: it is, why you think it isn't?

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java (hunk 2 above)
Comment (gmikerlemos) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:65]: less readable imho

Comment (ryandac) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:65]: no problem if you like it

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java (hunk 1 above)
Comment (ryandac) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:64]: Mike mentioned there would never be a HearingOption object with no hearing options set. If that is the case, we can resolve this comment.

**Review by ryandac — APPROVED** [review]

**Review by jackmaloney — APPROVED** [review]

**Pull request merged**